apps/system/js/pin_page_system_dialog.js

```diff
@@ -203,6 +215,24 @@
     SystemDialog.prototype.hide.apply(this);
   };
 
+  PinPageSystemDialog.prototype.updateHeight = function() {
+    var height = Service.query('LayoutManager.height');
+    this.containerElement.style.height = height + 'px';
+    // We need to manually set the gaia-dialog height,
+    // as it is fixed positioned
+    this.element.style.height = height + 'px';
+    this.debug('updateHeight: new height = ' + height);
+  };
+
+  SystemDialog.prototype.updateWidth = function sd_updateWidth() {
+    var width = Service.query('LayoutManager.width');
+    this.containerElement.style.width = width + 'px';
+    // We need to manually set the gaia-dialog width,
+    // as it is fixed positioned
```

Comment:
The whole problem about fixed and system dialogs is the SHB. 100% would mean the dialog overlaying the SHB in both portrait and landscape mode.